datafusion/functions/src/math/power.rs

```diff
@@ -104,45 +225,161 @@ impl ScalarUDFImpl for PowerFunc {
     fn invoke_with_args(&self, args: ScalarFunctionArgs) -> Result<ColumnarValue> {
         let args = ColumnarValue::values_to_arrays(&args.args)?;
```

Comment:
We could probably refactor this like so:

```rust
        let base = args.args[0].to_array(args.number_rows)?;
        let exponent = &args.args[1];
```

So we don't needlessly convert `args.arg[1]` from ColumnarValue (which may be Scalar) to Array back to ColumnarValue (now Array)